examples/MLIRGEMM/gemm.mlir

```diff
@@ -0,0 +1,289 @@
+#map0 = affine_map<(d0) -> (d0+10)>
+  
+
+#map10 = affine_map<(d0) -> (d0 * 64)> 
+#map11 = affine_map<(d0) -> (d0 * 64 + 64)>  
+#map12 = affine_map<(d0) -> (d0 * 256)>
+#map13 = affine_map<(d0) -> (d0 * 256 + 256)>  
+#map14 = affine_map<(d0) -> (d0 * 8)> 
+#map15 = affine_map<(d0) -> (d0 * 8 + 8)> 
+#map16 = affine_map<(d0) -> (d0 * 16)> 
+#map17 = affine_map<(d0) -> (d0 * 16 + 16)> 
+
+module{
+
+    func.func private @printMemrefF64(memref<*xf64>)
+
+    // naive muatrix multiplication using linalg dialect 
```

Comment:
typo `muatrix` -> `matrix`

Reply:
> @hxdtest Thanks for providing the examples! (1) Please reformat your code as @meshtag said above. (2) How do you measure the execution time? Can we add a GFLOPS report to the example? (see [here](https://github.com/buddy-compiler/buddy-mlir/blob/main/benchmark/conv2d-buddy-template#L69) for an example) (3) Have you considered adding a vectorization step to the example?

（1）I am confused about the format. Do we have some scripts like `pre-commit run` to check and reformat the code? 
（2）The GFLOPS report will be added in next commits.
  (3)   vectorization step will be added in next commits.

Reply:
> （1）I am confused about the format. Do we have some scripts like pre-commit run to check and reformat the code?

In my understanding, what @meshtag suggested is a changing of proper PR title and commit message. For example, you may change your commit message into something like `[examples][GEMM] add GEMM example`, using brackets to pointed out what part these commits are in. You may also provide more details in commit message like `add GEMM with different implementation` then lists them.

For code style in examples, we looks like don't have any automatically check yet, due to the lack of MLIR formatting tools. Code style checking of examples are usually done in PR manually. Your code style looks good for me for most of them, but I do notice some problems:

1. Unify the indents. Some of them are 4 spaces and others are 2, that's not good.
2. Giving reasons about these newlines. Normally one newline is enough, if in somewhere you use 2 or more of them, you may ask yourself "Why there is 2 or more newline?"
3. Files should be ended by a LF character.

BTW, I strongly suggest you to post this comment in PR rather than in my review comments, so more people may notice this. And the review comments will be folded after review is done, so it should only contains information related to the review only.

Reply:
Many thanks for detailed explanation. PR title is modified as you suggested and code style is fixed.